src/odemis/driver/pwrcomedi.py

```diff
@@ -124,33 +124,31 @@ def __init__(self, name, role, device, channels, spectra, pwr_curve, **kwargs):
                                      % (c, wl * 1e9))
             spect.append(tuple(wls))
 
-        max_power = max(crv[-1][1] for crv in self._pwr_curve)
-        # power of the whole device (=> max power of the device with max power)
-        self.power = model.FloatContinuous(0., (0., max_power), unit="W")
+        # Maximum power for channel to be used as a range for power
+        max_power = [crv[-1][1] for crv in self._pwr_curve]
+        # Power value for each channel of the device
+        self.power = model.ListContinuous(value=[0.] * len(self._channels),
+                                          range=([0.] * len(self._channels), max_power,), unit="W")
         self.power.subscribe(self._updatePower)
 
-        # ratio of power per channel
-        # => if some channel don't support max power, clamped before 1
-        self.emissions = model.ListVA([0.] * len(self._channels), unit="",
-                                      setter=self._setEmissions)
         # info on which channel is which wavelength
         self.spectra = model.ListVA(spect, unit="m", readonly=True)
 
         # make sure everything is off (turning on the HUB will turn on the lights)
-        self._updateIntensities(self.power.value, self.emissions.value)
+        self._updateIntensities(self.power.value)
 
         self._metadata = {model.MD_HW_NAME: self.getHwName()}
         lnx_ver = driver.get_linux_version()
         self._swVersion = "%s (driver %s, linux %s)" % (odemis.__version__,
-                                    self.getSwVersion(),
-                                    ".".join("%s" % v for v in lnx_ver))
+                                                        self.getSwVersion(),
+                                                        ".".join("%s" % v for v in lnx_ver))
         self._metadata[model.MD_SW_VERSION] = self._swVersion
-        self._metadata[model.MD_HW_VERSION] = self._hwVersion # unknown
+        self._metadata[model.MD_HW_VERSION] = self._hwVersion  # unknown
 
     def terminate(self):
         if self._device:
             # Make sure everything is powered off
-            self._updateIntensities(0, [0.] * len(self._channels))
+            self._updateIntensities([0.] * len(self._channels))
```

Comment:
You could also write `self._updateIntensities(self.power.range[0])`, which is slightly more explicit.